lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -244,37 +258,43 @@ putSignedTransaction (tx, witnesses) = do
 putWitness :: TxWitness -> Put
 putWitness witness =
     case witness of
-        PublicKeyWitness xPub (Hash sig) -> do
+        PublicKeyWitness _xPub (Hash sig) -> do
             -- Witness sum type:
-            --   * 1 for old address witnness scheme
-            --   * 2 for new address witness scheme
-            --   * 3 for account witness
+            --   * 0 for old address witnness scheme
+            --   * 1 for new address witness scheme
+            --   * 2 for account witness
+            --   * 3 for multisig
             putWord8 1
-            putByteString xPub
+            --putByteString xPub
             putByteString sig
         -- TODO: note that we are missing new address type witness:
         --  * https://github.com/input-output-hk/rust-cardano/blob/3524cfe138a10773caa6f0effacf69e792f915df/chain-impl-mockchain/doc/format.md#witnesses
         --  * https://github.com/input-output-hk/rust-cardano/blob/e0616f13bebd6b908320bddb1c1502dea0d3305a/chain-impl-mockchain/src/transaction/witness.rs#L23
         ScriptWitness _ -> error "unimplemented: serialize script witness"
         RedeemWitness _ -> error "unimplemented: serialize redeem witness"
 
-
 {-------------------------------------------------------------------------------
                             Common Structure
 -------------------------------------------------------------------------------}
 
 putTokenTransfer :: Tx -> Put
-putTokenTransfer (Tx inputs outputs) = do
+putTokenTransfer tx@(Tx inputs outputs) = do
     putWord8 $ fromIntegral $ length inputs
     putWord8 $ fromIntegral $ length outputs
+    putTxBody tx
+
+putTxBody :: Tx -> Put
+putTxBody (Tx inputs outputs) = do
     mapM_ putInput inputs
     mapM_ putOutput outputs
   where
     putInput (TxIn inputId inputIx) = do
         -- NOTE: special value 0xff indicates account spending
         -- only old utxo/address scheme supported for now
         putWord8 $ fromIntegral inputIx
+        putWord64be 14 -- TODO: We need input value!
```

Comment:
I am not sure I understand the idea behind this ?

EDIT: aha - I see https://github.com/input-output-hk/rust-cardano/blob/13af7aa08cdac8efc9dc8b0f7ff9162d28b2d479/chain-impl-mockchain/src/transaction/transfer.rs#L21 :+1: